types/events.go

```diff
@@ -74,23 +81,36 @@ func (em *EventManager) EmitTypedEvents(tevs ...proto.Message) error {
 
 // TypedEventToEvent takes typed event and converts to Event object
 func TypedEventToEvent(tev proto.Message) (Event, error) {
-	evtType := proto.MessageName(tev)
+	// Replace "." in event type with "-" to fix tm event query issue
+	evtType := strings.ReplaceAll(proto.MessageName(tev), ".", "-")
 	evtJSON, err := codec.ProtoMarshalJSON(tev, nil)
 	if err != nil {
 		return Event{}, err
 	}
 
-	var attrMap map[string]json.RawMessage
+	var attrMap map[string]interface{}
 	err = json.Unmarshal(evtJSON, &attrMap)
 	if err != nil {
 		return Event{}, err
 	}
 
 	attrs := make([]abci.EventAttribute, 0, len(attrMap))
 	for k, v := range attrMap {
+		var valueBz []byte
+		switch v := v.(type) {
+		case string:
+			valueBz = []byte(v)
```

Comment:
Is this special case necessary? What's the harm in JSON marshalling everything and then JSON unmarshalling?